WordPress/wordpress-playground
# Support users choosing how to handle URLs for sites that do not exist

## Motivation for the change, related issues

We want to be able to address specific Playgrounds via slug, even if the addressed Playground does not yet exist.

We'd rather not presume and store a Playground in a user's browser storage without their consent, so as a compromise, we can prompt the user and let them choose whether to store a site with the specified slug.

Related to #2037 cc @akirk 

## Implementation details

This PR adds support for a `if-stored-site-missing=prompt` query param that can be used alongside a `site-slug=<slug>` query param. If both params are provided and a site with the specified slug does not exist, we prompt the user and ask what they would like to do.

![Screenshot 2024-12-04 at 9 59 20 PM](https://github.com/user-attachments/assets/6f164297-8dbe-4b9a-8f84-bc3f29c947aa)

## Testing Instructions (or ideally a Blueprint)

- Run the dev server with `npm run dev`
- Visit http://127.0.0.1:5400/website-server/?if-stored-site-missing=prompt&site-slug=speed-build-challenge
- Confirm the modal displays and test that each button does as expected

Note that when choosing to continue with the temporary site, it is currently intentional that the original query params are not cleared. The reason is that if we are wanting to target an specific, possibly-non-existent slug, perhaps the user should be prompted with each page refresh.
